lib/dao.js

```diff
@@ -2356,6 +2566,184 @@ DataAccessObject.prototype.unsetAttribute = function unsetAttribute(name, nullif
 };
 
 /**
+ * Replace set of attributes.
+ * Performs validation before replacing.
+ *
+ * @trigger `validation`, `save` and `update` hooks
+ * @param {Object} data Data to replace
+ * @param {Object} [options] Options for replace
+ * @param {Function} cb Callback function called with (err, instance)
+ */
+
+DataAccessObject.prototype.replace = function replace(data, options, cb) {
```

Comment:
@Amir-61 please rename to `DataAccessObject.replaceById` (no `prototype`) as proposed by Raymond.